### PR TITLE
InstUI-2829 Allow to use node >= 12 on master

### PR DESCRIPTION
**Pull request by matyasf**
some dependencies stopped working with Node 12+, I've upgraded them
babel.config.js needs a new format for ES6 modules
`build:tokens` script broke because we specify what to export in `packagage.json`s, and this script needs to access stuff from other theme related packages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
     "yarn": "^1.17.3"
   },
   "engines": {
-    "node": "^10",
+    "node": ">=12.20",
```
Comment (matyasf) [inline package.json:90]: I have not tested with a lower minor version, it might work

File: packages/canvas-high-contrast-theme/package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
     ".": {
       "require": "./lib/index.js",
       "default": "./es/index.js"
-    }
+    },
+    "./package.json": "./package.json"
```
Comment (matyasf) [inline packages/canvas-high-contrast-theme/package.json:12]: This (and a few similar ones) are tricky. They cause an exception during build, but the build passes. Without adding these the `build:tokens` script breaks (this generates themes)

File: packages/canvas-theme/package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
     ".": {
       "require": "./lib/index.js",
       "default": "./es/index.js"
-    }
+    },
+    "./package.json": "./package.json"
```
Comment (matyasf) [inline packages/canvas-theme/package.json:12]: again, `build:tokens` fix

File: packages/ui-theme-tokens/package.json [hunk 4]
```diff
@@ -8,7 +8,11 @@
     ".": {
       "require": "./lib/index.js",
       "default": "./es/index.js"
-    }
+    },
+    "./lib/canvas": "./lib/canvas/index.js",
+    "./lib/canvasHighContrast": "./lib/canvasHighContrast/index.js",
+    "./lib/instructure": "./lib/instructure/index.js",
+    "./package.json": "./package.json"
```
Comment (matyasf) [inline packages/ui-theme-tokens/package.json:15]: again, `build:tokens` fix

**Review by HerrTopi — DISMISSED** [review]
I ran the code and it seems to be working. Nice job and it seems that a hard one as well :D

**Review by Brailor — APPROVED** [review]

**Review by Brailor — DISMISSED** [review]

**Review by Brailor — APPROVED** [review]

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**